planckton/sim.py

```diff
@@ -129,138 +129,134 @@ def __init__(
-                phase=0,
-                dynamic=["momentum"],
+            initial_box = sim.state.box
+            final_box = hoomd.Box(
+                Lx=final_length,
+                Ly=final_length,
+                Lz=final_length,
             )
-            log_quantities = [
-                "temperature",
-                "pressure",
-                "volume",
-                "potential_energy",
-                "kinetic_energy",
-                "pair_lj_energy",
-                "bond_harmonic_energy",
-                "angle_harmonic_energy",
-            ]
-            hoomd.analyze.log(
-                "trajectory.log",
-                quantities=log_quantities,
-                period=self.log_write,
-                header_prefix="#",
-                overwrite=False,
-                phase=0,
+            box_resize = hoomd.update.BoxResize(
+                box1=initial_box,
+                box2=final_box,
+                variant=ramp,
+                trigger=box_resize_trigger,
             )
-
-            if self.target_length is not None:
-                # Run the shrink step
-                final_length = self.target_length.to("Angstrom").value
-                final_box = (self.shrink_steps, final_length)
-                size_variant = hoomd.variant.linear_interp(
-                    [(0, snap.box.Lx), final_box], zero=0
-                )
-                box_resize = hoomd.update.box_resize(L=size_variant)
-                integrator.randomize_velocities(seed=42)
-                hoomd.run_upto(self.shrink_steps)
-                box_resize.disable()
-                self.n_steps = [i + self.shrink_steps for i in self.n_steps]
+            sim.operations.updaters.append(box_resize)
+            sim.state.thermalize_particle_momenta(
+                filter=all_particles, kT=self.shrink_kT
+            )
+            sim.run(self.shrink_steps, write_at_start=True)
+            self.n_steps = [i + self.shrink_steps for i in self.n_steps]
 
             # Begin temp ramp
             for kT, tau, n_steps in zip(self.kT, self.tau, self.n_steps):
-                integrator.set_params(kT=kT, tau=tau)
-                # Reset velocities
-                integrator.randomize_velocities(seed=42)
+                sim.operations.integrator.methods[0].kT = kT
+                sim.operations.integrator.methods[0].tau = tau
+                sim.state.thermalize_particle_momenta(
+                    filter=all_particles, kT=kT
+                )
 
-                try:
-                    hoomd.run_upto(n_steps + 1, limit_multiple=self.gsd_write)
-                    if sim.system.getCurrentTimeStep() >= self.n_steps[-1]:
-                        print("Simulation completed")
-                        done = True
-                except hoomd.WalltimeLimitReached:
-                    print("Walltime limit reached")
+                sim.run(n_steps)
+                if sim.timestep >= self.n_steps[-1]:
```

Comment:
Same comment here. 
For example, let's say `n_steps = [1000, 2000, 1000]`. At this line, `sim.timestep` is `4000 + shrink_steps`. Therefore, this if statement probably doesn't serve its purpose. 
You can check `sim.timestep > (sum of all items in n_step + shrink_steps`.